api/wasm/indigo-ketcher/indigo-ketcher.cpp

```diff
@@ -503,44 +501,50 @@ namespace indigo
             if (selected_atoms.size())
                 base += indigoCountAtoms(id);
         }
-
-        molecularWeightStream << VectorStringBracketed(molWeights);
-        mostAbundantMassStream << VectorStringBracketed(mamMasses);
-        monoisotopicMassStream << VectorStringBracketed(misoMasses);
-        massCompositionStream << VectorStringBracketed(massCompositions);
-        grossFormulaStream << VectorStringBracketed(grossFormulas);
     }
 
     void calculate_reaction(const IndigoKetcherObject& iko, std::stringstream& molecularWeightStream, std::stringstream& mostAbundantMassStream,
                             std::stringstream& monoisotopicMassStream, std::stringstream& massCompositionStream, std::stringstream& grossFormulaStream,
                             const std::vector<int>& selected_atoms)
     {
+        std::vector<std::string> molWeights[2], mamMasses[2], misoMasses[2], massCompositions[2], grossFormulas[2];
+
         const auto mol_iterator = IndigoObject(_checkResult(indigoIterateMolecules(iko.id())));
         while (const auto mol_id = _checkResult(indigoNext(mol_iterator.id)))
         {
             if (_checkResult(indigoCountRGroups(mol_id)) || _checkResult(indigoCountAttachmentPoints(mol_id)))
                 jsThrow("Cannot calculate properties for RGroups");
         }
         int base = 0;
-        calculate_iteration_object(IndigoObject(_checkResult(indigoIterateReactants(iko.id()))), molecularWeightStream, mostAbundantMassStream,
-                                   monoisotopicMassStream, massCompositionStream, grossFormulaStream, selected_atoms, base);
-
-        std::stringstream mws, mams, misos, mcs, gfs;
-        calculate_iteration_object(IndigoObject(_checkResult(indigoIterateProducts(iko.id()))), mws, mams, misos, mcs, gfs, selected_atoms, base);
-        if (grossFormulaStream.str().size() && gfs.str().size()) // check if we have gross formula for both reagents and products
-        {
+        calculate_iteration_object(IndigoObject(_checkResult(indigoIterateReactants(iko.id()))), molWeights[0], mamMasses[0], misoMasses[0], massCompositions[0], grossFormulas[0], selected_atoms, base);
+        calculate_iteration_object(IndigoObject(_checkResult(indigoIterateProducts(iko.id()))), molWeights[1], mamMasses[1], misoMasses[1], massCompositions[1], grossFormulas[1], selected_atoms, base);
+        
+        if (grossFormulas[0].size() && grossFormulas[1].size()) // check if we have gross formula for both reagents and products
+        {
+            molecularWeightStream << VectorStringBracketed(molWeights[0], true ); // true means we have both: products and reagents so all components are bracketed
```

Comment:
Please move the comment to separate line at the beginning of the block.
Also please add similar comment at the beginning of the else block.

Reply:
done